apps/checkin/views.py

```diff
@@ -0,0 +1,130 @@
+                return error_response("Must include uuid.",
+                                      "You must include the user's uuid from the bar code in the request body.",
+                                      400)
+            try:
+                group = CheckInGroup.objects.get(uuid=uuid)
+            except CheckInGroup.DoesNotExist:
+                return error_response("uuid does not exist",
+                                      "The uuid you specified does not exist in the database.",
+                                      404)
+            if group.checked_in:
+                return error_response("User already checked in.",
+                                      "The user specified is already checked in.",
+                                      409)
+            group.checked_in = True
+            group.save()
+            event = CheckInEvent(check_in_group=group, check_in=True, time=timezone.now())
+            event.save()
+            return success_data_jsonify()
+        else:
+            return error_response("Invalid method.", "Please use a post request.", 405)
+        # TODO: will need to be authenticated to admins
+
+
+class CheckOut(APIView):
+    permission_classes = (IsAdminUser,)
+    def post(self, request):
+        if request.POST:
+            if request.POST:
+                try:
+                    uuid = request.POST["uuid"]
+                except KeyError:
+                    return error_response("Must include uuid.",
+                                          "You must include the user's uuid from the bar code in the request body.",
+                                          400)
+                try:
+                    group = CheckInGroup.objects.get(uuid=uuid)
+                except CheckInGroup.DoesNotExist:
+                    return error_response("uuid does not exist",
+                                          "The uuid you specified does not exist in the database.",
+                                          404)
+                if not group.checked_in:
+                    return error_response("User is not checked in.",
+                                          "A user must be checked in to be checked out.",
+                                          409)
+                event = CheckInEvent(check_in_group=group, check_in=False, time=timezone.now())
+                group.checked_in = False
+                event.save()
+                group.save()
+                return success_data_jsonify()
+            else:
+                return error_response("Invalid method.", "Please use a post request.", 405)
+            # TODO: will need to be authenticated to admins
+
+
+# TODO: move the below functions to utils somewhere so everyone can use them
+def success_data_jsonify(obj={}, code=200):
+    response = JsonResponse({
+        'success' : True,
+        'data' : obj
+    })
+
+    response.status_code = code
+
+    return response
+
+
+def error_response(title, message, code):
```

Comment:
you should use the `Response` class from rest_framework instead of these classes - also you shouldn't need the success thing because the response code should tell you that - just return the result straight up if it works and return something like `{"detail": "User not found"}` on failure